sdk/communication/Azure.Communication.CallAutomation/src/CallConnection.cs

```diff
@@ -110,11 +110,10 @@ public virtual async Task<Response> HangUpAsync(HangUpOptions options, Cancellat
                 if (options == null)
                     throw new ArgumentNullException(nameof(options));
 
+                options.RepeatabilityHeaders?.GenerateIfRepeatabilityHeadersNotProvided();
```

Comment:
hang up call is already idempotent because it's a DELETE method so can move this under` if（options.ForEveryone)`. see Fang's design doc ref under `HangupCall`  section https://microsoft.sharepoint-df.com/teams/ServerCalling/_layouts/15/guestaccess.aspx?e=ml5jfd&share=EQ-3bH7vK3pBsos2I0l_4yoBgv7ak3VnUHudVSAFz6BPPQ

Reply:
I did not know this - thank you!